packages/asset-service/src/service/AssetService.ts

```diff
@@ -12,15 +18,45 @@ export const flattenAssetData = (assetData: BaseAsset[]): Asset[] => {
     flatAssetData.push(newAsset)
     if (baseAsset.tokens) {
       for (const tokenAsset of baseAsset.tokens) {
-        flatAssetData.push({
+        const flatAssetDatum = {
           ...tokenAsset,
           chain: baseAsset.chain,
           network: baseAsset.network,
           slip44: baseAsset.slip44,
           explorer: baseAsset.explorer,
           explorerAddressLink: baseAsset.explorerAddressLink,
           explorerTxLink: baseAsset.explorerTxLink
-        })
+        }
+        if (typeof flatAssetDatum.icon === 'undefined') {
```

Comment:
Is there any reason this doesn't work?
```suggestion
        if (!flatAssetDatum.icon) {
```

Reply:
Thanks for your suggestion!